R/convert_year_types.R

```diff
@@ -0,0 +1,46 @@
+#' Convert year types - Financial year form to the alternate form
+#' 
+#' Converts year type from the financial year form '1718' to the year form '2017'.
+#'
+#' @param fyyear vector of financial years in the form '1718'
+#'
+#' @return a vector of years in the alternate form '2017'
+#' @export
+#'
+#' @examples
+#' fyyears <- c("1718", "1819")
+#' convert_fyyear_to_year(fyyears)
+#' [1] "2017" "2018"
+convert_fyyear_to_year <- function(fyyear) {
+  for (i in 1:length(fyyear)) {
+    if (substr(fyyear[i], 1, 2) > substr(fyyear[i], 3, 4)) {
+      stop ("Year has been entered in the wrong format, try again using form `1718` or use function `convert_year_to_fyyear` to convert to the financial year form.")
+  }}
```

Comment:
You can get rid of the whole loop construct with `any()` i.e. if any of the supplied values are bad throw an error.
```suggestion
    if (any(substr(fyyear[i], 1, 2) > substr(fyyear[i], 3, 4))) {
      stop("Year has been entered in the wrong format, try again using form `1718` or use function `convert_year_to_fyyear` to convert to the financial year form.")
  }
```

We could also make the check more strict than just greater than, i.e. second part must be first part + 1:
```suggestion
  first_part <- as.integer(substr(fyyear, 1, 2))
  second_part <- as.integer(substr(fyyear, 3, 4))
  if (any(first_part != second_part + 1L)) {
    stop("Year has been entered in the wrong format, try again using form `1718` or use function `convert_year_to_fyyear` to convert to the financial year form.")
  }
```

Reply:
When I include your above code the test fails? but unsure how as all of the test years second part are +1 from the first

Reply:
Hmm, I'll have a quick look but I'm happy with it how it is too, I'll either push a change or approve shortly!